src/asynk/async_queue.rs

```diff
@@ -122,6 +149,38 @@ mod async_queue_tests {
         let result = queue.insert_task(&Job { number: 1 }).await.unwrap();
 
         assert_eq!(1, result);
+        queue.transaction.unwrap().rollback().await.unwrap();
+    }
+
+    #[tokio::test]
+    async fn remove_all_tasks_test() {
+        let pool = pool().await;
+        let mut connection = pool.get().await.unwrap();
+        let transaction = connection.transaction().await.unwrap();
+        let mut queue = AsyncQueue::<NoTls>::new_with_transaction(transaction);
+
+        let result = queue.insert_task(&Job { number: 1 }).await.unwrap();
+        assert_eq!(1, result);
+        let result = queue.insert_task(&Job { number: 2 }).await.unwrap();
+        assert_eq!(1, result);
+        let result = queue.remove_all_tasks().await.unwrap();
+        assert_eq!(2, result);
+        queue.transaction.unwrap().rollback().await.unwrap();
+    }
+    #[tokio::test]
+    async fn remove_tasks_type_test() {
+        let pool = pool().await;
+        let mut connection = pool.get().await.unwrap();
+        let transaction = connection.transaction().await.unwrap();
+        let mut queue = AsyncQueue::<NoTls>::new_with_transaction(transaction);
+
+        let result = queue.insert_task(&Job { number: 1 }).await.unwrap();
+        assert_eq!(1, result);
+        let result = queue.insert_task(&Job { number: 2 }).await.unwrap();
+        assert_eq!(1, result);
+        let result = queue.remove_tasks_type("common").await.unwrap();
+        assert_eq!(2, result);
+        queue.transaction.unwrap().rollback().await.unwrap();
```

Comment:
I think you don't need to roll back. it will be rolled back automatically when queue is dropped

Reply:
@ayrat555  It is true, you do not need to rollback explicitly but i think a rollback is an important operation that should be done explicitly, if you would to delete the explicit rollback,  i will delete them.